edb/server/compiler/errormech.py

```diff
@@ -139,24 +155,29 @@ def interpret_backend_error(schema, fields):
     column_name = fields.get('c')
     constraint_name = fields.get('n')
 
-    if code == PGError.NotNullViolationError:
-        source_name = pointer_name = None
-
-        if schema_name and table_name:
-            tabname = (schema_name, table_name)
+    return ErrorDetails(
+        message=message, detail=detail, detail_json=detail_json, code=code,
+        schema_name=schema_name, table_name=table_name,
+        column_name=column_name, constraint_name=constraint_name
+    )
 
-            source = common.get_object_from_backend_name(
-                schema, s_objtypes.ObjectType, tabname)
-            source_name = source.get_displayname(schema)
 
-            if column_name:
-                pointer_name = column_name
+def static_interpret_backend_error(fields):
+    err_details = get_error_details(fields)
+    if isinstance(err_details, errors.EdgeDBError):
+        return err_details
 
-        if pointer_name is not None:
-            pname = f'{source_name}.{pointer_name}'
+    message = err_details.message
+    detail = err_details.detail
+    detail_json = err_details.detail_json
+    code = err_details.code
+    schema_name = err_details.schema_name
+    table_name = err_details.table_name
+    constraint_name = err_details.constraint_name
```

Comment:
Just use `err_details` in the below code.  This packing/unpacking looks suspicious.

Reply:
OK